master-firmware/src/trace/trace.c

```diff
@@ -0,0 +1,145 @@
+#include "trace.h"
+#include <stddef.h>
+#include <string.h>
+
+#ifdef _CHIBIOS_RT_
```

Comment:
Or put it in a different translations unit and use functions instead of macros. I am not sure it would be significantly slower, especially if we start using LTO.